first-party/jib-spring-boot-extension-gradle/src/main/java/com/google/cloud/tools/jib/gradle/extension/springboot/JibSpringBootExtension.java

```diff
@@ -0,0 +1,123 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.gradle.extension.springboot;
+
+import com.google.cloud.tools.jib.api.JavaContainerBuilder;
+import com.google.cloud.tools.jib.api.buildplan.ContainerBuildPlan;
+import com.google.cloud.tools.jib.api.buildplan.FileEntriesLayer;
+import com.google.cloud.tools.jib.api.buildplan.FileEntry;
+import com.google.cloud.tools.jib.api.buildplan.LayerObject;
+import com.google.cloud.tools.jib.gradle.extension.GradleData;
+import com.google.cloud.tools.jib.gradle.extension.JibGradlePluginExtension;
+import com.google.cloud.tools.jib.plugins.extension.ExtensionLogger;
+import com.google.cloud.tools.jib.plugins.extension.ExtensionLogger.LogLevel;
+import com.google.cloud.tools.jib.plugins.extension.JibPluginExtensionException;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.UnknownTaskException;
+import org.gradle.api.tasks.TaskProvider;
+import org.springframework.boot.gradle.tasks.bundling.BootJar;
+
+public class JibSpringBootExtension implements JibGradlePluginExtension<Void> {
+
+  @Override
+  public Optional<Class<Void>> getExtraConfigType() {
+    return Optional.empty();
+  }
+
+  @Override
+  public ContainerBuildPlan extendContainerBuildPlan(
+      ContainerBuildPlan buildPlan,
+      Map<String, String> properties,
+      Optional<Void> config,
+      GradleData gradleData,
+      ExtensionLogger logger)
+      throws JibPluginExtensionException {
+    logger.log(LogLevel.LIFECYCLE, "Running Jib Spring Boot extension");
+
+    if (!shouldExcludeDevtools(gradleData.getProject(), properties, logger)) {
+      logger.log(LogLevel.INFO, "Keeping spring-boot-devtools (if any)");
+      return buildPlan;
+    }
+    logger.log(LogLevel.INFO, "Removing spring-boot-devtools (if any)");
+
+    List<LayerObject> newLayers =
+        buildPlan
+            .getLayers()
+            .stream()
+            .map(JibSpringBootExtension::filterOutDevtools)
+            .collect(Collectors.toList());
+    return buildPlan.toBuilder().setLayers(newLayers).build();
+  }
+
+  @VisibleForTesting
+  static boolean isDevtoolsJar(File file) {
+    return file.getName().startsWith("spring-boot-devtools-") && file.getName().endsWith(".jar");
+  }
+
+  @VisibleForTesting
+  static boolean shouldExcludeDevtools(
+      Project project, Map<String, String> extensionProperties, ExtensionLogger logger) {
+    try {
+      TaskProvider<Task> taskProvider = project.getTasks().named("bootJar");
```

Comment:
I think you can use a `find()` or something vs `named()` and avoid the exception?

Reply:
I don't know better than you, but the difference is returning `TaskProvider` vs `Task`. (Our main Jib code uses `named()`, FYI.) I just naively thought that `TaskProvider` might not "instantiate" or "trigger" something for projects that disabled or don't have `bootJar`.

Reply:
oh that's an interesting point, I didn't realize that that was the only option to get a taskProvider return value. I guess this is the way.